### PR TITLE
Custom styles

### PR DESCRIPTION
**Pull request by drewbo**
It works! But definitely open to changes in both the code and the verbose documentation. Let me know what you think @kelvinabrokwa 

Re: #112

screen shot 2015-12-09 at 4 11 53 pm

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -27,6 +27,29 @@ drawing | boolean | The ability to draw and delete features - default: `true`
 interactive | boolean | Keep all features permanently in edit mode - default: `false`
 keybindings | boolean | Keyboard shortcuts for drawing - default: `true`
 controls | Object | drawable shapes - default `{ marker: true, line: true, shape: true, square: true }`
+styles | Object | Add a style with any of the following properties: `gl-draw-polygon`, `gl-draw-polygon-stroke`, `gl-draw-line`, `gl-draw-point`, `gl-drawing-line`, `gl-drawing-polygon`, `gl-drawing-polygon-stroke`, `gl-drawing-points`, `gl-edit-line`, `gl-edit-polygon`, `gl-edit-polygon-stroke`, `gl-edit-point`, `gl-edit-point-mid`. The property should be an object with either the `layout` and/or `paint` properties as specified in the [Mapbox GL Style Reference](https://www.mapbox.com/mapbox-gl-style-spec/). It will overwrite the corresponding default  styles found in [`src/theme/`](https://github.com/mapbox/mapbox-gl-draw/tree/master/src/theme).
```
Comment (kelvinabrokwa) [inline API.md]: can you please make the properties into a bulleted list for the sake of readability?

**Comment by kelvinabrokwa** [comment]
@drewbo nice work! do you think its gl-draw's responsibility to validate the user's styles or should be just let gl-js throw on invalid styles?

I'd say option 2.

File: API.md (hunk 1 above)
Comment (drewbo) [inline API.md]: You don't like the list of 13 property names in a row?! (Will do)

**Comment by drewbo** [comment]
Thanks! Agree on option 2. This could also break other things if a user sends a style which has a `filters` or `source` property. Do we want to be "defensive" and prevent that?

**Comment by kelvinabrokwa** [comment]
@drewbo we can worry about that later. This is a great feature.

**Comment by drewbo** [comment]
Cool. This doesn't have any associated tests. Does it need anything? Otherwise I will squash the commits again.

**Comment by kelvinabrokwa** [comment]
@drewbo nah just the docs clean up and then I'll merge.

**Comment by drewbo** [comment]
Cool, I just hit the properties list thing

**Comment by kelvinabrokwa** [comment]
@drewbo duuuuudeeee, sorry to be so demanding, but markdown, please...

**Comment by drewbo** [comment]
I don't think there is a way to write markdown lists inside a table? If there is, happy to reformat

**Comment by kelvinabrokwa** [comment]
@drewbo I believe you. Thanks for the PR! Merging

**Pull request merged**